apps/frontend/src/components/ChessBoard.tsx

```diff
@@ -305,30 +322,53 @@ export const ChessBoard = memo(({
                                 to: squareRepresentation,
                               });
                             }
+                            const timeTaken =
+                              time - new Date(myMoveStartTime).getTime();
+                            const piece = chess.get(squareRepresentation)?.type;
                             if (moveResult) {
                               moveAudio.play();
 
                               if (moveResult?.captured) {
                                 captureAudio.play();
                               }
-                              setMoves((prev) => [...prev, moveResult]);
+                              setMoves((prev) => [
+                                ...prev,
+                                {
+                                  ...moveResult,
+                                  piece,
+                                  createdAt: myMoveStartTime,
+                                  timeTaken,
+                                },
+                              ]);
                               setFrom(null);
                               setLegalMoves([]);
                               if (moveResult.san.includes('#')) {
                                 setGameOver(true);
                               }
-                              socket.send(
-                                JSON.stringify({
-                                  type: MOVE,
-                                  payload: {
-                                    gameId,
-                                    move: moveResult,
-                                  },
-                                }),
-                              );
                             }
+                            socket.send(
+                              JSON.stringify({
+                                type: MOVE,
+                                payload: {
+                                  gameId,
+                                  move: {
+                                    from,
+                                    to: squareRepresentation,
+                                    san: moveResult?.san,
+                                    before: moveResult?.before,
+                                    after: moveResult?.after,
+                                    piece,
+                                    createdAt: myMoveStartTime,
```

Comment:
not a good idea, people can fake the time they did a move on

Reply:
Did in backend